pkg/build/controller/factory/factory.go

```diff
@@ -112,7 +112,7 @@ func (factory *BuildControllerFactory) CreateDeleteController() controller.Runna
 		RetryManager: controller.NewQueueRetryManager(
 			queue,
 			cache.MetaNamespaceKeyFunc,
-			limitedLogAndRetry(factory.BuildUpdater, 30*time.Minute),
+			controller.RetryNever,
```

Comment:
are you sure we don't want to retry on raised errors, now that NotFound errors don't propagate? what will clean up the build pod if not this?